### PR TITLE
Chore: Integration Tests

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

Have tests that check the main flows of the application with different edge cases and complex states.

# Changes

* Route level tests is allowed to mock only the api level. Test a main flow like creating a subaccount.

# Tests

All changes are in tests.

### CODE REVIEW

**Comment by lmuntaner** [comment]
@mstrasinskis @peterpeterparker please review

My idea would be to use component testing only for specific edge cases or complex behavior.

But otherwise we would rely on this high level tests that check that all components of a flow work together. Tests at this level allows us to prepare the state in the stores how we want it, then perform some steps, and check that the proper api actions were triggered.

Rule: routes/* and pages/* are allowed to mock **only** the api layer.

WDYT?

**Comment by lmuntaner** [comment]
@peterpeterparker @mstrasinskis please review

File: TESTING.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Testing
+
+Some background to understand and write tests for the nns-dapp.
```
Comment (peterpeterparker) [inline TESTING.md:3]: Add key word "frontend"?

`Some background to...frontend`

File: TESTING.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+Found in `frontend/src/tests/utils`.
+
+## Canister Test
+
+Tests are found in `rs/src/accounts_store/tests.rs`.
+
+### Running Tests
+
+```bash
+$ cd rs
```
Comment (peterpeterparker) [inline TESTING.md:53]: not sure but I think you can run these tests from root no?

**Review by peterpeterparker — APPROVED** [review]
LGTM, thx

File: TESTING.md (hunk 2 above)
Comment (lmuntaner) [reply TESTING.md:53]: I believe so as well, but to be consistent and for simplicity, I added the change of directory.
